python_modules/dagster/dagster_tests/core_tests/execution_tests/test_asset_backfill.py

```diff
@@ -710,7 +710,7 @@ def external_asset_graph_from_assets_by_repo_name(
 
         external_asset_nodes = external_asset_nodes_from_defs(
             repo.get_all_jobs(),
-            repo.assets_defs_by_key,
+            {k: ad for ad in repo.asset_graph.assets_defs for k in ad.keys},
```

Comment:
why not leave `assets_defs_by_key` as a useful shortcut? This seems unnecessarily thrashy?